links/src/test/java/juja/microservices/links/service/LinksServiceTest.java

```diff
@@ -1,7 +1,56 @@
 package juja.microservices.links.service;
 
-/**
- * @author Ivan Shapovalov
- */
+import juja.microservices.links.model.Link;
+import juja.microservices.links.model.SaveLinkRequest;
+import juja.microservices.links.repository.LinksRepository;
+import juja.microservices.links.service.impl.LinksServiceImpl;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
 public class LinksServiceTest {
+    @Rule
+    final public ExpectedException expectedException = ExpectedException.none();
+    private LinksService service;
+    @Mock
+    private LinksRepository repository;
+
+    @Before
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+        service = new LinksServiceImpl(repository);
+    }
+
+    @Test
```

Comment:
I don't understand this method.
We mock repository by Annotation @Mock, and then we test it?

Order of the tests is not pre-known. Second test can fail by NPE, if repo is null.

You can create mocks without runner and weird tests. Such as 
https://github.com/ivshapovalov/juja-platform/blob/a87ae67caf2186bc4db657cc93b7b912e31c4d4e/links-slackbot/src/test/java/juja/microservices/links/slackbot/service/LinksSlackbotServiceTest.java